selenium-system-test/src/test/scala/hts/pages/AuthorityWizardPage.scala

```diff
@@ -44,7 +50,24 @@ object AuthorityWizardPage extends WebPage {
     submit()
   }
 
-  def goToPage()(implicit driver: WebDriver): Unit =
+  def enterUserDetails(confidence: Int, credentialStrength: String, userInfo: TestUserInfo)(implicit driver: WebDriver): Unit = {
+    navigate()
+    setConfidenceLevel(confidence)
+    setCredentialStrength(credentialStrength)
+
+    setAddressLines(userInfo.address)
+
+    userInfo.address.postcode.foreach(setPostCode)
```

Comment:
can we use map instead of foreach on an Option ?

Reply:
yeah - `map` will basically do the same thing as `foreach` if the function you're using in `map` returns `Unit`. `foreach` is used for side-effecting functions which return `Unit` whereas `map is used to transform - this is why `foreach` returns `Unit` whereas `map` returns another `Option`. Here we're dealing with side-effecting functions which is why `foreach` is probably more appropriate here